src/docs/development/add-to-app/ios/project-setup.md

```diff
@@ -4,87 +4,95 @@ short-title: Integrate Flutter
-the module project that contains some CocoaPods and a helper Ruby script.
+Add your Dart code to the `lib/` directory.
+
+You can add Flutter dependencies to `my_flutter/pubspec.yaml`, including Flutter plugins 
+with iOS platform code.
+
+There is a `.ios/` hidden subfolder which contains a Xcode workspace where you can 
+run a stand-alone version of your module. It also contains helper scripts to embed
+the module into your existing application with [CocoaPods][].
 
-## Make the host app depend on the Flutter module
+{{site.alert.note}}
+Custom iOS code should be added to your existing application or a plugin, not
+the module in `.ios/`. Changes made in `.ios/` will not be embedded in your existing application.
+The directory can be regenerated by running `flutter clean` in the `my_flutter` directory.
+{{site.alert.end}}
 
-The description below assumes that your existing iOS app has a structure similar
-to what you get by asking Xcode to generate a new "Single View App" project
-using Objective-C. If your existing app has a different folder structure
-and/or existing `.xcconfig` files, you can reuse those, but you'll probably need
-to adjust some of the relative paths mentioned below accordingly.
+## Embed the Flutter module in your existing application
 
-The assumed folder structure is as follows:
+There are two ways to embed Flutter in your existing application.
+1. Use the CocoaPods dependency manager and `flutter`. Recommended.
+1. Create frameworks for the Flutter engine, your compiled Dart code, and all Flutter plugins.
+Manually embed the frameworks and update your existing application's build settings in Xcode.
+
+### Embed with CocoaPods and Flutter tools
+
+This method requires every developer working on your project to have `flutter` and [CocoaPods][] installed.
+
+The following example assumes your existing application and the Flutter module are in sibling directories.
+If you have a different directory organization, you may need to adjust the relative paths.
 
 ```text
 some/path/
   my_flutter/
-    lib/main.dart
-    .ios/
+    .ios/Flutter/podhelper.rb
   MyApp/
-    MyApp/
-      AppDelegate.h
-      AppDelegate.m (or swift)
-      :
+    Podfile
 ```
 
-### Add your Flutter app to your Podfile
-
-Integrating the Flutter framework requires using the CocoaPods dependency
-manager. This is because the Flutter framework needs to be available to any
-Flutter plugins that you might include in `my_flutter`.
-
-For information on how to install CocoaPods on your development machine, see
-[cocoapods.org](https://cocoapods.org/).
-
-If your host application (`MyApp`) is already using CocoaPods, you only have to do the
-following to integrate with your `my_flutter` app:
+If your existing application (`MyApp`) does not already have a Podfile,  follow the
+[CocoaPods getting started guide][] to add a `Podfile` to your project.
 
 1. Add the following lines to your `Podfile`:
 
 ```ruby
```

Comment:
See https://github.com/flutter/website/pull/3318/files. I think it's kinda nice to add a `<?code-excerpt "MyApp/Podfile" title?>` type thing that titles what file you're showing. But up to you.

Reply:
Good idea.